Sources/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -0,0 +1,215 @@
+        super.setUpLayout()
+
+        view.backgroundColor = appearance.colorPalette.background
+
+        messageListVC.view.translatesAutoresizingMaskIntoConstraints = false
+        addChildViewController(messageListVC, targetView: view)
+        messageListVC.view.pin(anchors: [.top, .leading, .trailing], to: view.safeAreaLayoutGuide)
+
+        messageComposerVC.view.translatesAutoresizingMaskIntoConstraints = false
+        addChildViewController(messageComposerVC, targetView: view)
+        messageComposerVC.view.pin(anchors: [.leading, .trailing], to: view)
+        messageComposerVC.view.topAnchor.pin(equalTo: messageListVC.view.bottomAnchor).isActive = true
+        messageComposerBottomConstraint = messageComposerVC.view.bottomAnchor.pin(equalTo: view.bottomAnchor)
+        messageComposerBottomConstraint?.isActive = true
+
+        NSLayoutConstraint.activate([
+            channelAvatarView.widthAnchor.pin(equalTo: channelAvatarView.heightAnchor),
+            channelAvatarView.heightAnchor.pin(equalToConstant: 32)
+        ])
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(customView: channelAvatarView)
+        channelAvatarView.content = (channelController.channel, client.currentUserId)
+
+        if let cid = channelController.cid {
+            headerView.channelController = client.channelController(for: cid)
+        }
+        
+        navigationItem.titleView = headerView
+        navigationItem.largeTitleDisplayMode = .never
+    }
+
+    override open func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        keyboardHandler.start()
+    }
+
+    override open func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        resignFirstResponder()
+
+        keyboardHandler.stop()
+    }
+}
+
+extension ChatChannelVC: ChatMessageListVCDataSource {
+    open func channel(for vc: ChatMessageListVC) -> ChatChannel? {
+        channelController.channel
+    }
+
+    open func numberOfMessages(in vc: ChatMessageListVC) -> Int {
+        channelController.messages.count
+    }
+
+    open func chatMessageListVC(_ vc: ChatMessageListVC, messageAt indexPath: IndexPath) -> ChatMessage? {
+        guard indexPath.item < channelController.messages.count else { return nil }
+        return channelController.messages[indexPath.item]
+    }
+
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
+        messageLayoutOptionsAt indexPath: IndexPath
+    ) -> ChatMessageLayoutOptions {
+        guard let channel = channelController.channel else { return [] }
+
+        return components.messageLayoutOptionsResolver.optionsForMessage(
+            at: indexPath,
+            in: channel,
+            with: AnyRandomAccessCollection(channelController.messages),
+            appearance: appearance
+        )
+    }
+}
+
+extension ChatChannelVC: ChatMessageListVCDelegate {
+    open func chatMessageListVC(
+        _ vc: ChatMessageListVC,
```

Comment:
if function has 2-3 arguments there is no need to declare them each on a new line
https://github.com/raywenderlich/swift-style-guide#function-declarations

>Keep short function declarations on one line including the opening brace:

Reply:
The team is well whereof the swift style guide btw 🙂 We try to follow when it makes sense, but we don't follow it religiously. 

Either way, in this case, the style guide doesn't mention that a function with 2-3 arguments should be in one line. It mentions "short" function declarations. A function can have 1 argument only hand be a long declaration. In this case, all function names in the delegate are pretty long, and so it makes sense to be in each line.

Reply:
okay, lets TBD on maxLineLenght and enforce it with a swiftlint config

Reply:
We already have that in place through **swiftformat** pre-commit hook in `hooks/pre-commit.sh`. The rules are in `.swiftformat` file and the max length is 132 `--maxwidth 132`